src/solvers/refinement/string_constraint_generator.h

```diff
@@ -40,14 +40,14 @@ class string_constraint_generatort final
   /// Arguments pack for the string_constraint_generator constructor
   struct infot
   {
-    const namespacet *ns=nullptr;
     /// Max length of non-deterministic strings
     size_t string_max_length=std::numeric_limits<size_t>::max();
     /// Prefer printable characters in non-deterministic strings
     bool string_printable=false;
   };
 
-  explicit string_constraint_generatort(const infot& info);
+  explicit string_constraint_generatort(
+    const infot& info, const namespacet& ns);
```

Comment:
The other `infot` structs contain all needed constructor arguments. An exception is warranted here because `namespacet` in `string_constraint_generatort` is used only for debug printing, whereas in other classes it has a "real" use. Hence, in order to avoid inheritance namespace collisions with `infot`s, `namespacet` is excluded.